homeassistant/components/axis.py

```diff
@@ -0,0 +1,347 @@
+"""
+Support for Axis devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/axis/
+"""
+
+import json
+import logging
+import os
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import discovery
+
+from homeassistant.const import (ATTR_LOCATION, ATTR_TRIPPED,
+                                 CONF_HOST, CONF_INCLUDE, CONF_NAME,
+                                 CONF_PASSWORD, CONF_SCAN_INTERVAL,
+                                 CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.helpers.entity import Entity
+from homeassistant.loader import get_component
+
+
+# REQUIREMENTS = ['axis==5']
+REQUIREMENTS = ['http://github.com/Kane610/axis'
+                '/archive/v5.zip'
+                '#axis==5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'axis'
+CONFIG_FILE = 'axis.conf'
+
+AXIS_DEVICES = {}
+
+EVENT_TYPES = ['motion', 'vmd3', 'pir', 'sound',
+               'daynight', 'tampering', 'input']
+
+PLATFORMS = ['camera']
+
+AXIS_INCLUDE = EVENT_TYPES + PLATFORMS
+
+AXIS_DEFAULT_NAME = 'Axis network device'
+AXIS_DEFAULT_HOST = '192.168.0.90'
+AXIS_DEFAULT_USERNAME = 'root'
+AXIS_DEFAULT_PASSWORD = 'pass'
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_INCLUDE):
+        vol.All(cv.ensure_list, [vol.In(AXIS_INCLUDE)]),
+    vol.Optional(CONF_NAME, default=AXIS_DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_HOST, default=AXIS_DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_USERNAME, default=AXIS_DEFAULT_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD, default=AXIS_DEFAULT_PASSWORD): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=0): cv.positive_int,
+    vol.Optional(ATTR_LOCATION, default=None): cv.string,
+}, extra=vol.ALLOW_EXTRA)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.slug: DEVICE_SCHEMA,
+    }),
+}, required=False, extra=vol.ALLOW_EXTRA)
+
+
+def request_configuration(hass, name, host, serialnumber):
+    """Request configuration steps from the user."""
+    configurator = get_component('configurator')
+
+    def configuration_callback(callback_data):
+        """Called when config is submitted."""
+        if CONF_INCLUDE not in callback_data:
+            configurator.notify_errors(instance,
+                                       'Functionality mandatory.')
+            return False
+        callback_data[CONF_INCLUDE] = callback_data[CONF_INCLUDE].split()
+        callback_data[CONF_HOST] = host
+        if CONF_NAME not in callback_data:
+            callback_data[CONF_NAME] = name
+        try:
+            config = DEVICE_SCHEMA(callback_data)
+        except:  # pylint: disable=W0702
```

Comment:
Indicate the most specific exception type in except block, instead of just except